apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -201,3 +256,86 @@ func validateRootDeviceHints(rdh *RootDeviceHints) error {
 	}
 	return nil
 }
+
+// When making changes to this function for operationalstatus and errortype,
+// also make the corresponding changes in the OperationalStatus and
+// ErrorType fields in the struct definition of BareMetalHostStatus in
+// the file baremetalhost_types.go
+func checkStatusAnnotation(bmhstatus *BareMetalHostStatus) error {
+
+	operatstatus_allowed := []string{"", string(OperationalStatusOK), string(OperationalStatusDiscovered), string(OperationalStatusError), string(OperationalStatusDelayed), string(OperationalStatusDetached)}
+	errortype_allowed := []string{"", string(ProvisionedRegistrationError), string(RegistrationError), string(InspectionError), string(PreparationError), string(ProvisioningError), string(PowerManagementError)}
```

Comment:
I suggest you move these to where the constants are defined so that we don't forget to update them.

Reply:
moved these to the types file alongside the declaration of its element type